provider-profile-service/src/main/java/team/proximity/provider_profile_service/payment_method/PaymentMethodRepository.java

```diff
@@ -15,10 +15,16 @@ public interface PaymentMethodRepository extends JpaRepository<PaymentMethod, Lo
 
     Optional<PaymentMethod> findByIdAndCreatedBy(Long id, String createdBy);
 
+    @Query("SELECT p FROM PaymentMethod p WHERE p.createdBy = " + ":createdBy AND p.paymentPreference = :paymentPreference")
```

Comment:
why the concatenation of the queries?

Reply:
The concatenation in the query is unnecessary and was likely a mistake. I'll fix it by directly using the parameter without concatenation for clarity and correctness

Reply:
fixed